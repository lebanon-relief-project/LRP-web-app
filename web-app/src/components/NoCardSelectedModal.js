import React from "react";
import styled from "styled-components";
import CloseButton from "../assets/images/CloseButton.svg";
import devices from "../styles/Devices";
import colours from "../styles/Colours";

const NoCardSelectedModal = (props) => {
  return (
    <ModalWrapper>
      <Card>
        <StyledCloseButton
          src={CloseButton}
          alt="Close Button"
          onClick={props.closeModal}
        />
        <Title>Looks like you haven't selected any cards</Title>
        <Text>
          We can help with tailored advice if you select at least one card, by
          giving it a green tick.
        </Text>
        <StyledButton onClick={props.closeModal}>OK</StyledButton>
      </Card>
    </ModalWrapper>
  );
};

const Title = styled.h2`
  font-family: Playfair Display;
  font-style: normal;
  font-size: 30px;
  line-height: 38px;
  color: ${colours.black};
`;

const Text = styled.p`
  font-family: Raleway;
  font-weight: 400;
  font-style: normal;
  font-size: 16px;
  line-height: 24px;
`;

const StyledButton = styled.button`
  cursor: pointer;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px 16px;
  width: 100%;
  background: ${colours.yellow};
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.043);
  border-radius: 2px;
`;

const ModalWrapper = styled.div`
  background: rgba(67, 67, 67, 0.6);
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 200;
`;

const Card = styled.div`
  position: absolute;
  left: 28.06%;
  right: 28.06%;
  top: 38.28%;
  background: ${colours.white};
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.25);
  border-radius: 6px;
  padding: 3%;
  @media (max-width: ${devices.ipadpro}) {
    left: 15%;
    right: 15%;
  }
  @media (max-width: ${devices.mobile}) {
    left: 10%;
    right: 10%;
  }
`;

const StyledCloseButton = styled.img`
  cursor: pointer;
  position: absolute;
  top: 15%;
  right: 5%;
  width: 4%;
  height: auto;
  @media (max-width: ${devices.mobile}) {
    width: 6%;
    height: auto;
    top: 10%;
    right: 8%;
  }
`;

export default NoCardSelectedModal;
