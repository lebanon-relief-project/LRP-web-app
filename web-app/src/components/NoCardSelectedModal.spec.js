import React from "react";
import { render } from "@testing-library/react";
import NoCardSelectedModal from "./NoCardSelectedModal";
import "jest-styled-components";
import { fireEvent } from "@testing-library/dom";
import { act } from "react-dom/test-utils";

describe("the NoCardSelectedModal component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderNoCardSelectedModal());

    expect(container).toMatchSnapshot();
  });
  it("should close Modal when close button is clicked", () => {
    const closeModal = jest.fn();
    let getByAltText;
    ({ getByAltText } = render(
      <NoCardSelectedModal closeModal={closeModal} />
    ));

    act(() => {
      fireEvent.click(getByAltText("Close Button"));
    });

    expect(closeModal).toHaveBeenCalled();
  });
});

const renderNoCardSelectedModal = () => {
  return render(<NoCardSelectedModal />);
};
