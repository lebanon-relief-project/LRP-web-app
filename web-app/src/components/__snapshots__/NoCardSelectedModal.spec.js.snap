// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the NoCardSelectedModal component should match the snapshot 1`] = `
.c3 {
  font-family: Playfair Display;
  font-style: normal;
  font-size: 30px;
  line-height: 38px;
  color: #000000;
}

.c4 {
  font-family: Raleway;
  font-weight: 400;
  font-style: normal;
  font-size: 16px;
  line-height: 24px;
}

.c5 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 16px;
  width: 100%;
  background: #FFD666;
  box-shadow: 0px 2px 0px rgba(0,0,0,0.043);
  border-radius: 2px;
}

.c0 {
  background: rgba(67,67,67,0.6);
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 3;
}

.c1 {
  position: absolute;
  left: 28.06%;
  right: 28.06%;
  top: 38.28%;
  background: #ffffff;
  box-shadow: 0px 1px 3px rgba(0,0,0,0.25);
  border-radius: 6px;
  padding: 3%;
}

.c2 {
  cursor: pointer;
  position: absolute;
  top: 15%;
  right: 5%;
  width: 4%;
  height: auto;
}

@media (max-width:1025px) {
  .c1 {
    left: 15%;
    right: 15%;
  }
}

@media (max-width:576px) {
  .c1 {
    left: 10%;
    right: 10%;
  }
}

@media (max-width:576px) {
  .c2 {
    width: 6%;
    height: auto;
    top: 10%;
    right: 8%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Close Button"
        class="c2"
        src="CloseButton.svg"
      />
      <h2
        class="c3"
      >
        Looks like you haven't selected any cards
      </h2>
      <p
        class="c4"
      >
        We can help with tailored advice if you select at least one card, by giving it a green tick.
      </p>
      <button
        class="c5"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
